qiskit_metal/analyses/quantization/lumped_capacitive.py

```diff
@@ -733,6 +737,50 @@ def load_q3d_capacitance_matrix(path, user_units='fF', _disp=True):
     return df_cmat, user_units, design_variation, df_cond
 
 
+def lumped_oscillator_from_path(path: str, Lj_nH: float, Cj_fF: float, N: int,
+                                fr: Union[list, float],
+                                fb: Union[list, float]) -> pd.DataFrame:
+    """Obtain a single result dataframe from a Q3D capacitance file pointed to by path.
+    Similar to member method lumped_oscillator_vs_passes() of QQ3DRenderer but for a user
+    provided capacitance matrix file.
+
+    Args:
+        path (str): Path to file.
+        Lj_nH (float): Junction inductance (in nH)
+        Cj_fF (float): Junction capacitance (in fF)
+        N (int): Coupling pads (1 readout, N - 1 bus)
+        fr (Union[list, float]):Readout frequencies (in GHz). fr can be a list with the order
+            they appear in the capMatrix.
+        fb (Union[list, float]): Coupling bus frequencies (in GHz). fb can be a list with the order
+            they appear in the capMatrix.
+        g_scale (float, optional): Scale factor. Defaults to 1..
+
+    Returns:
+        dict: A single dataframe corresponding to a single capacitance matrix
+    """
+    IC_Amps = Convert.Ic_from_Lj(Lj_nH, 'nH', 'A')
+    CJ = ureg(f'{Cj_fF} fF').to('farad').magnitude
+    fr = ureg(f'{fr} GHz').to('GHz').magnitude
+    fb = [ureg(f'{freq} GHz').to('GHz').magnitude for freq in fb]
+
+    df_cmat, user_units, _, _, = load_q3d_capacitance_matrix(path)
```

Comment:
Might want to add an error check incase the file isn't correct, right format, etc.

Reply:
That being said, this method q3d-specific, thus incompatible with other simulation tools (necessarily, given the specific file-format). Thus a key decision to make is: should we put every method that is related to a specific renderer/tool into the renderer itself? (thus should this method (And any other q3d-specific method) be moved into the renderer? (or static-methods file within the same folder of the ansys renderers?)

Reply:
@quantum-quant , glad you think similarly. if @ThomasGM4 has nothing against it, maybe we can do that in a separate PR (moving all of the q3d methods around could be troublesome. no need to hold off this PR).

Reply:
Sure can have as a separate PR